File: mykube.js
function kube() {
	

 
  var i = document.getElementById("input").value
  var stat = "0"
  var final = ""
  if (((i.includes("deployments")) || (i.includes("deploy")) || (i.includes("deployment"))) && ((i.includes("create")) || (i.includes("generate")) || (i.includes("build")))) {
    var image = prompt("Enter the name of image: ")
	var deployment_name = prompt("Enter the name of Deployment: ")
	var stat = "1"
	var final = stat + " " + image + " " + deployment_name 
	}

  else if (((i.includes("pods")) || (i.includes("pod"))) && ((i.includes("create")) || (i.includes("generate")) || (i.includes("build")))) {
	var image = prompt("Enter the name of image: ")
	var pod_name = prompt("Enter the name of pod: ")
	var stat = "2"
	var final = stat + " " + image + " " + pod_name
	
	}

  else if (((i.includes("pods")) || (i.includes("pod"))) && ((i.includes("delete")) || (i.includes("remove")) || (i.includes("detach")))) {
	var pod_name = prompt("Enter the name of pod: ")
	var stat = "3"
	var final = stat + " " + pod_name
	
	}

  else if (((i.includes("deployments")) || (i.includes("deploy")) || (i.includes("deployment"))) && ((i.includes("delete")) || (i.includes("remove")) || (i.includes("detach")))) {
	var deployment_name = prompt("Enter the name of Deployment: ")
	var stat = "4"
	var final = stat + " " + deployment_name
	}

  else if (((i.includes("deployments")) || (i.includes("deploy")) || (i.includes("deployment"))) && ((i.includes("expose"))  || (i.includes("display")))) {
	var deployment_name = prompt("Enter the name of Deployment: ")
	var port_no = prompt("Enter the port number where you want to export your deployment: ")
	var expose_type = prompt(" 1. ClusterIP \n 2.ExternalNames \n 3.LoadBalancer 4.NodePort \n Enter the type which you want to export ")
	var stat = "5"
	var final = stat + " " + deployment_name + " " + port_no + " " + expose_type
	}

  else if (((i.includes("deployments")) || (i.includes("deploy")) || (i.includes("deployment"))) && ((i.includes("scale")) || (i.includes("increase")) || (i.includes("decrease")) || (i.includes("scale up")) || (i.includes("scale down")))) {
	var deployment_name = prompt("Enter the name of Deployment: ")
	var replica = prompt("Enter number of replicas: ")
	var stat = "6"
	var final = stat + " " + deployment_name + " " + replica
	}

  else if (((i.includes("pods")) || (i.includes("pod"))) && ((i.includes("show")) || (i.includes("get")) || (i.includes("list")))) {
	var stat = "7"
	var final = stat 
	
	}

  else if (((i.includes("deployments"))|| (i.includes("deployment")) || (i.includes("deploy"))) && ((i.includes("show")) || (i.includes("get")) || (i.includes("list")))) {
	var stat = "8"
	var final = stat 
	
	}
  else if (((i.includes("service")) || (i.includes("services")) || (i.includes("svc"))) && ((i.includes("show")) || (i.includes("get")) || (i.includes("list")))) {
	var stat = "9"
	var final = stat
	
	}

  else if (((i.includes("all")) || (i.includes("everything"))) && ((i.includes("delete")) || (i.includes("remove")) || (i.includes("destroy")))) {
	var stat = "10"
	var final = stat 
	}

  else {
	var stat = "11"
	var final = stat 
	}

	
  var xhr = new XMLHttpRequest();
  xhr.open("GET", "http://localhost/cgi-bin/mykube.py?k= " + final, "true")
  xhr.send();
  
  xhr.onload = function ()
	{
	document.getElementById('out3').innerHTML = xhr.responseText; 
        }
	
  
}

File: mykube.py
#!/usr/bin/python3

import cgi
import subprocess as sp

print("content-type: text/html")
print()

mydata = cgi.FieldStorage()
data= mydata.getvalue("k")

var = data.split()
if var[0] == "1":
    depname = var[2]
    image_name = var[1]
    cmd = "kubectl create deployment " + (depname) + " --image=" + (image_name)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig admin.conf")
    print(output)
elif var[0] == "2":
    pod = var[2]
    image_name = var[1]
    cmd = "kubectl run " + (pod) + " --image=" + (image_name)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig admin.conf")
    print(output)
elif var[0] == "3":
    pod = var[1]
    cmd = "kubectl delete pod " + (pod_name)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig admin.conf")
    print(output)
elif var[0] == "4":
    depname = var[1]
    cmd = "kubectl delete deployment " + (depname)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig admin.conf")
    print(output)
elif var[0] == "5":
    depname = var[1]
    port = var[2]
    Expose =  var[3]
    cmd = "kubectl expose deployment " + (depname) + " --port=" + (port) + " --type=" + (Expose)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig admin.conf")
    print(output)
elif var[0] == "6":
    depname = var[1]
    replica = var[2]
    cmd = "kubectl scale deployment " + (depname) + " --replicas=" + (replica)
    output = sp.getoutput("sudo " + cmd + " --kubeconfig  admin.conf")
    print(output)
elif var[0] == "7":
    cmd = "sudo kubectl get pods --kubeconfig admin.conf"
    output = sp.getoutput(cmd)
    print(output)
elif var[0] == "8":
    cmd = "kubectl get deployments --kubeconfig admin.conf"
    output = sp.getoutput("sudo " + cmd)
    print(output)
elif var[0] == "9":
    cmd = "kubectl get svc --kubeconfig admin.conf"
    output = sp.getoutput(cmd)
    print(output)
elif var[0] == "10":
    cmd = "kubectl delete all --all --kubeconfig admin.conf"
    output = sp.getoutput("sudo " + cmd)
    print(output)
elif var[0] == "11":
    print("Hey . Give it another try")
